.github/workflows/build.yaml

```diff
@@ -11,6 +11,10 @@ env:
   REGISTRY: ghcr.io
 
 jobs:
+  clear-cache:
+    if: github.event_name == 'schedule'
+    uses: ./.github/workflows/cache.yaml
+
```

Comment:
I guess as a separate job this is going to run in parallel with the build job. Perhaps it should be a step that always runs before build if it should be triggered so the build is clean when it should be.

Reply:
The cache needs to be cleared before building the asdf plugins for tests as well. Maybe another `nightly.yaml` workflow that is clearing cache and then triggering the `main.yaml` workflow would be better. With a separate workflow you wouldn't need the `if: github.event_name == 'schedule'`

Reply:
resolved by splitting `nightly.yaml`